modules/nextflow/src/main/groovy/nextflow/processor/TaskContext.groovy

```diff
@@ -303,17 +307,28 @@ class TaskContext implements Map<String,Object>, Cloneable {
         if( !path )
             throw new ProcessException("Process `$name` missing template name")
 
-        if( !(path instanceof Path) )
+        if( path !instanceof Path )
             path = Paths.get(path.toString())
 
         // if the path is already absolute just return it
         if( path.isAbsolute() )
             return path
 
-        // otherwise make
+        // make from the module dir
+        def module = NF.isDsl2Final() ? ScriptMeta.get(this.script)?.getModuleDir() : null
+        if( module ) {
+            def target = module.resolve('templates').resolve(path)
+            if (Files.exists(target))
+                return target
+        }
+
+        // otherwise make from the base dir
         def base = Global.session.baseDir
-        if( base )
-            return base.resolve('templates').resolve(path)
+        if( base ) {
+            def target = base.resolve('templates').resolve(path)
+            if (Files.exists(target))
+                return target
+        }
 
```

Comment:
I would keep the old version ie. returning the file path even if not existing to avoid unexpected side effects with legacy code.

Reply:
Sorry, I didn't mean to break backward compatibility. Just fixed.